x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -262,44 +262,55 @@ export class TaskRunner<
 
     let updatedAlertTypeState: void | Record<string, unknown>;
     try {
-      updatedAlertTypeState = await this.alertType.executor({
-        alertId,
-        services: {
-          ...services,
-          alertInstanceFactory: createAlertInstanceFactory<
-            InstanceState,
-            InstanceContext,
-            WithoutReservedActionGroups<ActionGroupIds, RecoveryActionGroupId>
-          >(alertInstances),
-        },
-        params,
-        state: alertTypeState as State,
-        startedAt: this.taskInstance.startedAt!,
-        previousStartedAt: previousStartedAt ? new Date(previousStartedAt) : null,
-        spaceId,
-        namespace,
-        name,
-        tags,
-        createdBy,
-        updatedBy,
-        rule: {
+      const ctx = {
+        type: 'alert',
+        name: `execute ${alert.alertTypeId}`,
+        id: alertId,
+        description: `execute ${alert.alertTypeId} with ${name} in ${namespace ?? 'default'}`,
+        // is there a way to create a URL to navigate to an alert
+        // url: '',
+      };
+
+      updatedAlertTypeState = await this.context.executionContext.withContext(ctx, () =>
```

Comment:
heh, I thought this was going to involve annotating ES and/or SO clients, but I guess it's just simple magic! :-)

So now the question is, how can I test this - jest test and/or function test.  About to go looking at the code ...

Reply:
> So now the question is, how can I test this - jest test and/or function test.

That's a good question. I have a task to write integration tests for the functionality, but realistically I can start work on it after the FF date only.
We can brainstorm how to approach this testing. We could either leverage Elasticsearch [slow logs](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules-slowlog.html) to perform kinda e2e testing. Or test the Kibana functionality by piping Kibana logs to a local file and run assertions against it.